# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../mine/x-no-cksums.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

To reduce redundancy our reduncancy detector found that this goold
old Un*x chksums are not worth much and unused despite the size field.
We now only use the md5sums files where we added the otherwise missing
size tag.

  - Rene Rebe <rene@exactcode.de>

diff -urN mine-0.23-orig/create.c mine-0.23/create.c
--- mine-0.23-orig/create.c	2005-03-23 09:51:06.000000000 +0100
+++ mine-0.23/create.c	2006-10-29 20:19:46.000000000 +0100
@@ -1,6 +1,7 @@
 /*
  *  GEM MINE - The ROCK Linux Package Manager
  *  Copyright (C) 2002-2005 Clifford Wolf
+ *  Copyright (C) 2006 Rene Rebe
  *  
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -99,7 +100,6 @@
 	"packages",
 	"flists",
 	"md5sums",
-	"cksums",
 	"dependencies",
 	"descs",
 	NULL
diff -urN mine-0.23-orig/md5sum.c mine-0.23/md5sum.c
--- mine-0.23-orig/md5sum.c	2006-10-29 20:16:56.000000000 +0100
+++ mine-0.23/md5sum.c	2006-10-29 20:19:20.000000000 +0100
@@ -70,8 +70,8 @@
 	struct dirent *md5_dent;
 	char buffer[1024], buffer2[1024];
 	char *md5sum, *filename;
-	const char const *admdirs[6] = 
-	{"cksums", "dependencies", "descs", "flists", "md5sums", "packages"};
+	const char const *admdirs[5] = 
+	{"dependencies", "descs", "flists", "md5sums", "packages"};
 	FILE *f;
 	DIR *d;
 
@@ -90,7 +90,7 @@
 			/* Add the /var/adm files of each package to the 
 			   md5sum memdb so md5sum_check() recognizes them. */
 			int n;
-			for (n = 0; n < 6; n++) {
+			for (n = 0; n < 5; n++) {
 				snprintf(buffer2, 1024, "var/adm/%s/%s",
 					admdirs[n], md5_dent->d_name);
 				memdb_put(&md5_memdb, buffer2, 
diff -urN mine-0.23-orig/mine.c mine-0.23/mine.c
--- mine-0.23-orig/mine.c	2005-09-30 12:34:36.000000000 +0200
+++ mine-0.23/mine.c	2006-10-29 20:19:51.000000000 +0100
@@ -1,6 +1,7 @@
 /*
  *  GEM MINE - The ROCK Linux Package Manager
  *  Copyright (C) 2002-2005 Clifford Wolf
+ *  Copyright (C) 2006 Rene Rebe
  *  
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -98,7 +99,7 @@
 "    Where {query-mode} might be one of:\n"
 "\n"
 "       -q  name and version    -p  long description    -l  file list\n"
-"       -m  md5sums             -c  cksums              -d  dependencies\n"
+"       -m  md5sums             -d  dependencies\n"
 "\n"
 "       -k key  (for *.gem files only, use 'LIST' as key to list all keys)\n"
 "       -P      create 'packages.db' records (for *.gem files only)\n"
@@ -145,7 +146,6 @@
 "        packages        The /var/adm/packages/<pkg> file\n"
 "        flists          The /var/adm/flists/<pkg> file\n"
 "        md5sums         The /var/adm/md5sums/<pkg> file\n"
-"        cksums          The /var/adm/cksums/<pkg> file\n"
 "        dependencies    The /var/adm/dependencies/<pkg> file\n"
 "        descs           The /var/adm/descs/<pkg> file\n"
 "\n"
@@ -247,10 +247,6 @@
 				set_mode(MINE_MODE_SHOWFILE);
 				mine_file_key_name = "md5sums";
 				break;
-			case 'c':
-				set_mode(MINE_MODE_SHOWFILE);
-				mine_file_key_name = "cksums";
-				break;
 			case 'd':
 				set_mode(MINE_MODE_SHOWFILE);
 				mine_file_key_name = "dependencies";
diff -urN mine-0.23-orig/mkpdb.c mine-0.23/mkpdb.c
--- mine-0.23-orig/mkpdb.c	2005-03-23 09:51:06.000000000 +0100
+++ mine-0.23/mkpdb.c	2006-10-29 20:19:56.000000000 +0100
@@ -1,6 +1,7 @@
 /*
  *  GEM MINE - The ROCK Linux Package Manager
  *  Copyright (C) 2002-2005 Clifford Wolf
+ *  Copyright (C) 2006 Rene Rebe
  *  
  *  This program is free software; you can redistribute it and/or modify
  *  it under the terms of the GNU General Public License as published by
@@ -71,7 +72,6 @@
 	if ( dump_field(&c, "pkg_name") == -1 ) goto error;
 	if ( dump_field(&c, "descs") == -1 ) goto error;
 	if ( dump_field(&c, "dependencies") == -1 ) goto error;
-	if ( dump_field(&c, "cksums") == -1 ) goto error;
 
 	putchar('\004');
 	putchar('\n');
diff -urN mine-0.23-orig/readdb.c mine-0.23/readdb.c
--- mine-0.23-orig/readdb.c	2006-10-29 20:16:56.000000000 +0100
+++ mine-0.23/readdb.c	2006-10-29 20:17:15.000000000 +0100
@@ -256,7 +256,7 @@
 	}
 
 	if ( !dbf ) {
-		snprintf(filename, PATH_MAX, "%s/%s/info/cksums/%s",
+		snprintf(filename, PATH_MAX, "%s/%s/info/md5sums/%s",
 			sourcedir, config, packagename);
 		f = fopen(filename, "r");
 	} else {
